src/react/PipelineSourcePage.tsx

```diff
@@ -57,30 +50,30 @@ export default ({ astarte, history, pipelineId }) => {
 
   let innerHTML;
 
-  switch (phase) {
+  switch (pipelineFetcher.status) {
```

Comment:
here the code is a bit complex, but we could try using the `WaitForData` component and rely on the fallback for the loading spinner

Reply:
You're right. I've now used WaitForData in both PipelineSource and Pipelines pages